src/Symfony/Component/Security/Http/Authenticator/AbstractFormLoginAuthenticator.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator;
+
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Http\EntryPoint\AuthenticationEntryPointInterface;
+
+/**
+ * A base class to make form login authentication easier!
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ *
+ * @experimental in 5.1
+ */
+abstract class AbstractFormLoginAuthenticator extends AbstractAuthenticator implements AuthenticationEntryPointInterface, RememberMeSupportedInterface, InteractiveAuthenticatorInterface
```

Comment:
We could rename this to `AbstractLoginFormAuthenticator`. I'm trying to avoid naming conflicts with Guard classes when possible/reasonable. We got lucky that I used the name `AbstractGuardAuthenticator` so that our new `AbstractAuthenticator` doesn't conflict.